Core/Src/DynamicControls/longitudinal_control.c

```diff
@@ -31,3 +39,7 @@ uint32_t longitudinal_control(uint8_t wheel_motor){
 
 
 };
+void longitudinal_control(uint32_t *torque_decrease_longitudinal){
+    torque_decrease_longitudinal[R_MOTOR] = wheel_control(R_MOTOR);
+    torque_decrease_longitudinal[L_MOTOR] = wheel_control(L_MOTOR);
+}
```

Comment:
então, não era isso q eu tava pensando ainda rsrs
```suggestion
// define essa struct no .h
typedef struct {
    uint32_t torque_decrease[2];
} longitudinal_control_result_t;
...
longitudinal_control_result_t longitudinal_control() {
    longitudinal_control_result_t result;
    // logica do torque_decrease_longitudinal vem aq
    ...
    result.torque_decrease[R_MOTOR] = PID_compute(...);
    result.torque_decrease[L_MOTOR] = PID_compute(...);
    
    return result;
}
...
// la no torque_manager.c

longitudinal_control_result_t longitudinal_control_result = longitudinal_control();
...
```
Ou então recebe essa struct por referencia e popula ela. Mas se vc fazer isso, n adianta nada criar uma instância toda vez q vai chamar a função, igual tá la no código da torque_manager.c. Pra justificar a passagem por referencia, vc precisa fazer essa instancia ter um escopo maior e ser persistente durante múltiplas chamadas do loop da tarefa de torque_manager, se n vc perde todas as vantagens da otimização

Reply:
vc n acha que seria positivo manter a função wheel_control, para melhorar o encapsulamento do codigo? Se  eu fizer a logica do PID toda no longitudinal control fica poluido e pouco modularizado.

Reply:
na verdade eu concordo com vc, o que tava me incomodando é pq vc vai acessar a mesma variável global duas vezes num espaço mt curto de tempo (= péssimo para performance). Masss vc consegue melhorar isso: pode deixar assim do jeito q vc tá falando, mas aí adiciona um parâmetro na função de wheel_control pra receber o valor da variável global (por referência) e pega os dados dele.